core/slice/slice.odin

```diff
@@ -117,6 +131,61 @@ linear_search_proc :: proc(array: $A/[]$T, f: proc(T) -> bool) -> (index: int, f
 	return -1, false
 }
 
+/*
+	Reverse linear search searches the given element in the given slice in O(n) time,
+	starting from the slice end.
+
+	Returns the last index at which the given element can be found in the slice,
+	or -1 if it is not present
+
+	# Examples
+
+	```
+	index: int
+	found: bool
+
+	a := []i32{1, 1, 1, 2}
+
+	index, found = linear_search_reverse(a, 2)
+	assert(index == 3 && found == true)
+
+	index, found = linear_search_reverse(a, 1)
+	assert(index == 2 && found == true)
+
+	index, found = linear_search_reverse(a, 0)
+	assert(index == -1 && found == false)
+	```
+
+	If you need a custom compare procedure, see `linear_search_reverse_proc`
+*/
+@(require_results)
+linear_search_reverse :: proc(array: $A/[]$T, key: T) -> (index: int, found: bool)
+	where intrinsics.type_is_comparable(T) #no_bounds_check {
```

Comment:
`#no_bounds_check` isn't needed, none of the operations add bounds checks (same thing with the reverse proc).

Reply:
What about plain `linear_search`? It does use `#no_bounds_check`

Reply:
I see, I am also pretty sure it isn't needed there.